highfidelity/hifi
# FB18302 Remove deprecated Package Model functionality

This PR updates Interface's `Package Model as .fst...` to eliminate deprecated functionality.

Packaging models now works as-if the "Avatar Body and Head" type were always selected (beyond UI changes there should be no impact on overall application behavior or avatar use).

The main user-facing change is that Package Model => `Model Type:` dropdown and its secondary options are no more:
* _Avatar Body and Head_ (now the internal default)
* _Avatar Attachment_ (removed)
* _Entity Model_ (removed)

Change summary:
* Remove `Model Type: ` dropdown.
* Rename `Package Model as .fst...` to `Package Avatar as .fst...`
* Remove non-functional Qt "whatsthis" dialog icon.
* Eliminate unneeded code paths per above.

#### Testing:
##### "A" testing:
1) Run current production Beta Release.
2) Selected "Edit -> Package Model as .fst" 
3) Choose an .fbx file (leaving the Model Type dropdown set to "Avatar Body with Head").
4) Pick reasonable joint options / hit OK to package the model.
5) ... take note of the generated FST and asset folders.

##### "B" testing:
1) Run this PR build.
2) Selected "Edit -> Package Avatar as .fst" 
3) Choose an .fbx file.
4) Pick reasonable joint options / hit OK to package the model.
5) ... take note of the generated FST and asset folders.

Compare A and B packaged output -- assuming identical FBX and joint options were selected then the generated FST files and folder contents should be identical too.

https://highfidelity.fogbugz.com/f/cases/18302
